### PR TITLE
Create BSTtoMinHeap

### PR DESCRIPTION
**Pull request by deepti-iiitr**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
#7095

### CODE REVIEW

**Comment by deepti-iiitr** [comment]
@HarshCasper @ankitaggarwal23 Can you please review my PR?

File: C-Plus-Plus/ds/BSTtoMinHeap.cpp [hunk 1]
```diff
@@ -0,0 +1,155 @@
+// Program to convert the given Binary Search Tree into a min heap, given that BST is a complete binary tree
+
+
+#include <iostream>
+#include <vector>
+#include <queue>
+#include <string>
+#include <utility>
```
Comment (siddharth25pandey) [inline C-Plus-Plus/ds/BSTtoMinHeap.cpp]: Use single header **bits/stdc++**

File: C-Plus-Plus/ds/BSTtoMinHeap.cpp [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    inorder(root, keys);
+ 
+    // traverse tree in preorder fashion, and for each encountered node,
+    // dequeue a key and assign it to the node
+    preorder(root, keys);
+}
+ 
+int main()
+{
+    vector<int> keys = { 5, 3, 2, 4, 8, 6, 10 };
```
Comment (siddharth25pandey) [inline C-Plus-Plus/ds/BSTtoMinHeap.cpp]: Use Dynamic Input

File: C-Plus-Plus/ds/BSTtoMinHeap.cpp [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    for (int key: keys) {
+        root = insert(root, key);
+    }
+ 
+    convert(root);
+    printLevelOrderTraversal(root);
+ 
+    return 0;
+}
+
```
Comment (siddharth25pandey) [inline C-Plus-Plus/ds/BSTtoMinHeap.cpp:125]: Write Time and Space Complexity for the given Program

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@deepti23-web Do the suggested changes

**Comment by siddharth25pandey** [comment]
@deepti23-web Also add Update in README file

File: C-Plus-Plus/ds/BSTtoMinHeap.cpp (hunk 1 above)
Comment (deepti-iiitr) [reply C-Plus-Plus/ds/BSTtoMinHeap.cpp]: I have added the single header file.

File: C-Plus-Plus/ds/BSTtoMinHeap.cpp (hunk 2 above)
Comment (deepti-iiitr) [reply C-Plus-Plus/ds/BSTtoMinHeap.cpp]: I have edited this file to use dynamic inputs.

File: C-Plus-Plus/ds/BSTtoMinHeap.cpp (hunk 3 above)
Comment (deepti-iiitr) [reply C-Plus-Plus/ds/BSTtoMinHeap.cpp:125]: I wrote the time and space complexity.

**Comment by deepti-iiitr** [comment]
@siddharth25pandey I have made the required changes. Kindly look into this.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Use proper code Format 
- Remove extra lines of code and spaces
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add a new line at the EOF
- Use proper display message how you are reading and writing outputs
- Add input-output prompts to make the program more interactive.

**Comment by deepti-iiitr** [comment]
@ankitaggarwal23 I have made the required changes. Kindly see.

File: C-Plus-Plus/ds/BSTtoMinHeap.cpp [hunk 4]
```diff
@@ -0,0 +1,125 @@
+// Program to convert the given BST into a Min Heap.
+// Conditions are that all the values in the left subtree of a node should be less than all the values in the right subtree of the node.
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/BSTtoMinHeap.cpp]: Use multiline comments

File: C-Plus-Plus/ds/BSTtoMinHeap.cpp [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        root = insert(root, key);
+    }
+    convert(root);
+    printLevelOrderTraversal(root);
+    return 0;
+}
+// Sample Input:
+// 7
+// 4 2 6 1 3 5 7
+// The time complexity of the above program is O(n) and the space complexity is also O(n).
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/BSTtoMinHeap.cpp]: Remove this and rewrite it using the following format
```
/*

Sample Input/Output:

Input:

Output:


Time complexity:
Space complexity:

*/
```
Write the outputs in the way your code is producing.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Rest LGTM💯

File: C-Plus-Plus/ds/BSTtoMinHeap.cpp (hunk 4 above)
Comment (deepti-iiitr) [reply C-Plus-Plus/ds/BSTtoMinHeap.cpp]: Done.

File: C-Plus-Plus/ds/BSTtoMinHeap.cpp (hunk 5 above)
Comment (deepti-iiitr) [reply C-Plus-Plus/ds/BSTtoMinHeap.cpp]: Done. Kindly see.

**Comment by siddharth25pandey** [comment]
> @deepti23-web Also add Update in README file

@deepti23-web Not done yet

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Write the outputs in the way your code is producing.

**Comment by deepti-iiitr** [comment]
@ankitaggarwal23 I have written the output in the same way. Do you mean, that "Enter the no. of node" and "Enter the no. of keys" should also be added in the file?

**Comment by ankitaggarwal23** [comment]
> @ankitaggarwal23 I have written the output in the same way. Do you mean, that "Enter the no. of node" and "Enter the no. of keys" should also be added in the file?

Exactly and update the readme.md. if you find any issue related to readme just have a look into some merged PR you will get some idea. If you still facing some issue do let me know will help you.

**Comment by deepti-iiitr** [comment]
> > @deepti23-web Also add Update in README file
> 
> @deepti23-web Not done yet

Done now.

**Comment by deepti-iiitr** [comment]
@ankitaggarwal23 @siddharth25pandey I have done the required changes and updated Readme file. Kindly review my PR.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**